scripts/getting_started.sh

```diff
@@ -0,0 +1,583 @@
+        ## check  if denpency is installed successfully
+        isCmdInstalled  "${cmdName}"
+        exitIfNot $? 0 "Installing ${cmdName} failed, exit..."
+    fi
+}
+
+
+function intallDependencies(){
+    installDocker
+    enableDocker
+    installKind
+    installClusterctl
+    commonInstall jq "sudo apt install -y jq"
+    commonInstall kubectl "sudo snap install kubectl --classic"
+    commonInstall make "sudo apt install -y build-essential"
+    commonInstall go "sudo snap install go --classic"
+}
+
+function commonInstall(){
+    cmdName=$1
+    installCmd=$2
+
+    ## check  if denpency is installed before
+    isCmdInstalled "${cmdName}"
+
+    ## install denpency if it not installed
+    if [ $? -eq 0 ] ; then
+        runCmd "${installCmd}" 0 "Installing ${cmdName}..."
+
+        ## check  if denpency is installed successfully
+        isCmdInstalled  "${cmdName}"
+        exitIfNot $? 0 "Installing ${cmdName} failed, exit..."
+    fi
+}
+
+function createKindCluster(){
+    echo  "Creating kind cluster..."
+
+    msg=$(kind create cluster 2>&1)
+    if [ $? -ne 0 ]; then
+        echo "Create kind cluster failed"
+        echo $msg | grep -q "You have reached your pull rate limit"
+        if [ $? -eq 0 ]; then 
+            echo "Suggestion: you can use \"docker login\" to avoid such an error."
+            exit
+        fi
+    else
+        echo "Create kind cluster successfully"
+    fi
+}
+
+function downloadByohCode(){
+    runCmd "rm -rf cluster-api-provider-bringyourownhost" 0 " Cleaning byoh code..."
+    msg=$(git clone git@github.com:vmware-tanzu/cluster-api-provider-bringyourownhost.git 2>&1)
+    if [ $? -ne 0 ]; then
+        echo "Downloading byoh code..."
+        echo $msg | grep "Please make sure you have the correct access rights"
+        if [ $? -eq 0 ]; then 
+            echo "Suggestion: Add an public key of this machine into \"SSH and GPG keys\" of your github setting"
+            exit
+        fi
+    else
+        echo "Download byoh code successfully"
+    fi
+}
+
+export PATH=/snap/bin:${PATH}
+intallDependencies 
+ 
+runCmd "sudo swapoff -a" 0 "Turning off swap..."
+swapMsg=$(sudo swapon -s)
+if [ -n "${swapMsg}" ]; then
+    echo "Please turn off swap first."
+    exit
+fi
+ 
+# check if cluster "kind" is already exited
+clusterName=`kind get clusters`
+if [ "${clusterName}" != "kind" ]; then
+    #runCmd "kind create cluster" 0 "Creating kind cluster..."
+    createKindCluster
+fi
+
+configClusterctl
+
+
+# check if init it before
+kubectl get pods --all-namespaces | grep -q byoh-controller-manager
+if [ $? -ne 0 ]; then
+    runCmd "clusterctl init --infrastructure byoh" 0 " Transforming the Kubernetes cluster into a management cluster..."
+else
+    echo "clusterctl init --infrastructure byoh before"
+fi
+
+#check if byoh image is existed
+docker images | grep "byoh-dev/node" | grep -q "v1.22.3"
```

Comment:
same here. We should just run the make command to build the docker image

Reply:
@huchen2021 This is a good idea. @dharmjit & I have also thought about this in the past. Instead of having this as a release artifact, we should push this image to Harbor and we can pull it from there. This will be useful in e2e test as well.

Reply:
@anusha94 Ok. I'll fix this after we have  this image to Harbor.

Reply:
@huchen2021 ,
Please try this image -
`docker pull projects.registry.vmware.com/cluster_api_provider_bringyourownhost/byoh/node-ubuntu-20.04:e2e`